shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/profile/orders/presentation/OrderHistoryComponent.kt

```diff
@@ -11,36 +11,46 @@ import com.vml.tutorial.plantshop.profile.orders.presentation.states.OrderHistor
 import com.vml.tutorial.plantshop.profile.orders.presentation.states.OrderListItemUiState
 import com.vml.tutorial.plantshop.profile.orders.data.OrdersRepository
 import com.vml.tutorial.plantshop.profile.orders.data.usecase.OrderPlantsUseCase
+import com.vml.tutorial.plantshop.profile.orders.domain.OrderItem
 import com.vml.tutorial.plantshop.profile.orders.domain.OrderStatus
+import com.vml.tutorial.plantshop.profile.orders.presentation.states.OrderHistoryCommonUiState
 import com.vml.tutorial.plantshop.profile.orders.presentation.states.OrderHistoryConfirmAction
+import kotlinx.coroutines.async
+import kotlinx.coroutines.awaitAll
 import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted
 import kotlinx.coroutines.flow.StateFlow
-import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.stateIn
 import kotlinx.coroutines.flow.update
 import kotlinx.coroutines.launch
 
-class OrderHistoryComponent(
+open class OrderHistoryComponent(
     componentContext: ComponentContext,
-    private val ordersRepository: OrdersRepository,
-    private val plantsRepository: PlantsRepository,
+    internal val ordersRepository: OrdersRepository,
+    internal val plantsRepository: PlantsRepository,
     private val orderPlants: OrderPlantsUseCase = OrderPlantsUseCase(ordersRepository),
     private val onComponentEvent: (event: OrderHistoryEvents) -> Unit
 ): ComponentContext by componentContext  {
-    private val _uiState: MutableStateFlow<OrderHistoryUiState> = MutableStateFlow(OrderHistoryUiState())
-    val uiState: StateFlow<OrderHistoryUiState> = _uiState.asStateFlow()
+    internal val commonUiState: MutableStateFlow<OrderHistoryCommonUiState> = MutableStateFlow(OrderHistoryCommonUiState())
 
-    init {
-        fetchOrderHistory()
-    }
+    private val _uiState: MutableStateFlow<OrderHistoryUiState> = MutableStateFlow(OrderHistoryUiState())
+    val uiState: StateFlow<OrderHistoryUiState> = commonUiState.combine(_uiState) { common, ui ->
+        ui.copy(commonState = common)
+    }.stateIn(componentCoroutineScope(), SharingStarted.WhileSubscribed(), _uiState.value)
 
-    private fun fetchOrderHistory() {
+    open fun fetchOrderHistory() {
         componentCoroutineScope().launch {
             showLoader()
-            val orders = ordersRepository.getOrders()
+            val items = listOf(OrderStatus.PENDING, OrderStatus.SHIPPED, OrderStatus.CANCELLED).map { status ->
+                componentCoroutineScope().async {
+                    ordersRepository.getOrders(status = status, limit = 3)
+                }
+            }.awaitAll().flatten()
+
             _uiState.update {
                 it.copy(
-                    contentLoading = false,
-                    items = orders.map { order ->
+                    items = items.map { order ->
                         OrderListItemUiState(
```

Comment:
I think it would be better to extract this part to a function in order to reduce the number of indentations here. You don't need to change it but it would look prettier in my opinion.

Reply:
Yes you are right thanks